internal/lang/references.go

```diff
@@ -83,5 +87,23 @@ func ReferencesInExpr(parseRef ParseRef, expr hcl.Expression) ([]*addrs.Referenc
 		return nil, nil
 	}
 	traversals := expr.Variables()
+	if fexpr, ok := expr.(hcl.ExpressionWithFunctions); ok {
+		funcs := filterProviderFunctions(fexpr.Functions())
+		traversals = append(traversals, funcs...)
+	}
 	return References(parseRef, traversals)
 }
+
+func filterProviderFunctions(funcs []hcl.Traversal) []hcl.Traversal {
+	pfuncs := make([]hcl.Traversal, 0, len(funcs))
+	for _, fn := range funcs {
+		if len(fn) > 0 {
+			if root, ok := fn[0].(hcl.TraverseRoot); ok {
+				if strings.HasPrefix(root.Name, "provider::") {
```

Comment:
Nit: we might want to pull out this check to a functon, we use it in many places.